pkg/cmd/provisioning/gcp/create_service_accounts.go

```diff
@@ -264,8 +262,15 @@ func createServiceAccount(ctx context.Context, client gcp.Client, name string, c
 				}
 			} else {
 				log.Printf("Existing IAM custom role %s found, updating permissions", role.Title)
-				if !actuator.AreSlicesEqualWithoutOrder(role.IncludedPermissions, gcpProviderSpec.Permissions) {
-					role.IncludedPermissions = gcpProviderSpec.Permissions
+				addedPermissions, removedPermissions := actuator.CalculateSliceDiff(role.IncludedPermissions, gcpProviderSpec.Permissions)
+
+				if len(removedPermissions) > 0 {
+					allRemovedPermissions := strings.Join(removedPermissions, ", ")
+					log.Printf("The following permissions for IAM custom role %s will NOT be removed: [%s] (permissions deltas can only be additive)", role.Title, allRemovedPermissions)
```

Comment:
Reword the log... something along the lines of "Unexpected permissions found on existing custom role..." etc.